### PR TITLE
XAI ODH WG proposal

### PR DESCRIPTION
**Pull request by tteofili**
Explainability for ODH Working Group proposal.

## Description

This is a proposal for a working group around Explainability (aka XAI) in the context of ODH.

## Open points
It's still to be decided who the `approvers` and `reviewers` would be (in `wg/OWNERS`)

## How Has This Been Tested?

No testing is needed AFAIK.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by taneem-ibrahim** [comment]
@anishasthana Could you please help review this?

File: wg-xai/OWNERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+reviewers:
+  -
```
Comment (anishasthana) [inline wg-xai/OWNERS:2]: We should include a reviewers/approvers group. At a minimum a list of people will be needed.

File: wg-xai/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Organizers
+
+* Rebecca Whitworth
+* Tommaso Teofili
+* Rui Vieira
+* Rob Geada
+
+## Contact
+- Slack: [#wg-ai-explainabilityt-framework](https://odh-io.slack.com/archives/C03UFCVFFEY)
```
Comment (anishasthana) [inline wg-xai/README.md:28]: The slack channel is now called `wg-ai-explainability` right?

**Review by anishasthana — COMMENTED** [review]
Couple of small changes.

@LaVLaS Do we need to explicitly call out a SIG under which this WG would be running?

**Comment by LaVLaS** [comment]
> Couple of small changes.
> 
> @LaVLaS Do we need to explicitly call out a SIG under which this WG would be running?

The MLOps and DevEx SIGS are referenced as stakeholders in the README

File: wg-xai/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# AI Explainability Working Group
+
+A Working Group dedicated to establishing (designing, developing and documenting) an explainability framework for ODH developers (based on TrustyAI explainability toolkit). 
+Explanations are useful tools in order to:
+Establish trust in predictive models created / used in ODH, especially when deployed in decision making pipelines, e.g. provide explanations together with predictions to end users to allow humans to take better informed decisions
+Debug predictions made by predictive models created / used in ODH, e.g. detect (portions of) model input that are responsible for counterintuitive predictions
+Get a global understanding of general behavior of predictive models, e.g. which features does a model mostly attend to?
+
+## Stakeholder SIGs
+* [SIG ML Developer Experience](/sig-ml-developer-experience)
```
Comment (anishasthana) [inline wg-xai/README.md:10]: Should we just remove the Dev Ex SIG from here and scope the WG to establishing the backend components for explainability?

File: wg-xai/README.md (hunk 2 above)
Comment (tteofili) [reply wg-xai/README.md:28]: yes, correct. I'll fix that.

File: wg-xai/README.md (hunk 3 above)
Comment (tteofili) [reply wg-xai/README.md:10]: we can do that, I'd probably keep the SIG ML ops as stakelholder, WDYT?

File: wg-xai/OWNERS (hunk 1 above)
Comment (tteofili) [reply wg-xai/OWNERS:2]: I'll fix that.

File: wg-xai/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+## Goals
+
+* Establish a framework for the inclusion of explainability algorithms/capabilities in the Open Data Hub ecosystem in a cohesive and unified manner.
+* Provide tools to ODH developers for explaining predictions made by predictive models.
+* Provide tools to ODH developers for global model explainability. 
+
+## Deliverables
+
+* Provide TrustyAI-enabled JupiterHub extension for ODH.
```
Comment (anishasthana) [inline wg-xai/README.md:39]:
```suggestion
* Provide TrustyAI-enabled JupyterHub extension for ODH.
```

**Review by anishasthana — COMMENTED** [review]
One last change, and can you squash the commits as well?

File: wg-xai/OWNERS [hunk 5]
```diff
@@ -0,0 +1,6 @@
+reviewers:
+  - rui, tteofili, dzonca, rgeada, tibrahim
```
Comment (anishasthana) [inline wg-xai/OWNERS:2]:
```suggestion
  - dzonca
  - rgeada
  - rui
  - tibrahim
  - tteofili
```

File: wg-xai/OWNERS [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+approvers:
+  - rui, tteofili, dzonca, rgeada, tibrahim
```
Comment (anishasthana) [inline wg-xai/OWNERS:4]:
```suggestion
  - dzonca
  - rgeada
  - rui
  - tibrahim
  - tteofili
```

File: wg-xai/README.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+* [TrustyAI Blog](https://blog.kie.org/category/all?s=trustyai)
+* [An introduction to Explainable AI](https://www.youtube.com/watch?v=mg_4UvQzC3w)
+* [KIELive#51 - TrustyAI: Ensuring the Fairness and Transparency of Decision Models](https://www.youtube.com/watch?v=C5NGczQMHu0)
+* [Shapley Additive Explanations (SHAP)](https://www.youtube.com/watch?v=VB9uV-x0gtg)
+* [Introduction to Counterfactuals and how it helps understanding black-box prediction models](https://www.youtube.com/watch?v=zXwbzo_GcwA)
+* [Explanation by Example: the OptaPlanner way](https://www.youtube.com/watch?v=4H3U6xyCgMI)
+* [TrustyAI Drools Integration](https://drive.google.com/file/d/1Ti8HOzKUIJw7Qzp7J4jHEB0uYUd6lhYs/view)
+* [TrustyAI Explainability Toolkit paper](https://arxiv.org/abs/2104.12717)
+
+
```
Comment (anishasthana) [inline wg-xai/README.md:63]: Extra new line here.

**Review by anishasthana — COMMENTED** [review]
Can you folks also propose this working group in our next SIG meeting? We actually have one coming up on December 14th at 14:00 Eastern Time.

**Comment by tteofili** [comment]
@anishasthana we can propose this WG in the next SIG meeting for sure.
In the meantime I've updated the PR, please let me know if there's anything left to fix.

**Review by anishasthana — APPROVED** [review]

**Pull request merged**